bpf/gotracer/go_nethttp.c

```diff
@@ -828,17 +842,23 @@ int beyla_uprobe_http2RoundTrip(struct pt_regs *ctx) {
     void *goroutine_addr = GOROUTINE_PTR(ctx);
     void *cc_ptr = GO_PARAM1(ctx);
 
-    setup_http2_client_conn(goroutine_addr, cc_ptr);
+    setup_http2_client_conn(
+        goroutine_addr, cc_ptr, _cc_tconn_pos, _cc_next_stream_id_pos, _cc_framer_pos);
 
     return 0;
 }
 
+// For the vendored version of http2 inside the Go runtime, when they upgrade HTTP to HTTP2.
 SEC("uprobe/http2RoundTripConn")
 int beyla_uprobe_http2RoundTripConn(struct pt_regs *ctx) {
     void *goroutine_addr = GOROUTINE_PTR(ctx);
     void *cc_ptr = GO_PARAM1(ctx);
 
-    setup_http2_client_conn(goroutine_addr, cc_ptr);
+    setup_http2_client_conn(goroutine_addr,
```

Comment:
The offsets per versions are different and a binary can have both vendored and non-vendored Go. We must supply which probe offsets to use in the helper routine.